src/main/java/seedu/address/logic/commands/PermaSortContactCommand.java

```diff
@@ -0,0 +1,140 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Comparator;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+
+
+/**
+ * Sorts the contacts in Athena's address book permanently.
+ * Index entered determines the specific order.
+ */
+public class PermaSortContactCommand extends Command {
+
+    public static final String COMMAND_WORD = "psort";
+
+    public static final String COMMAND_TYPE = CommandType.CONTACT.toString();
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sort your contacts permanently "
+            + "by the index command entered  "
+            + "1 will be sort by alphabetical order of their names\n"
+            + "2 will be sort by alphabetical order of their address\n"
+            + "3 will be sort by alphabetical order of their first tag\n"
+            + "Parameters: INDEX (must be between 1 and 3) "
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    public static final String MESSAGE_INVALID_INDEX = "Invalid index entered, "
+            + "refer to below for the command's proper usage: "
+            + MESSAGE_USAGE;
+
+    private static final Comparator<Person> NAME_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+        }
+    };
+
+    private static final Comparator<Person> ADDRESS_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            return o1.getAddress().value.compareToIgnoreCase(o2.getAddress().value);
+        }
+    };
+
+    private static final Comparator<Person> TAG_COMPARATOR = new Comparator<Person>() {
+        @Override
+        public int compare(Person o1, Person o2) {
+            if (o1.getTags().size() == 0 && o2.getTags().size() != 0) {
+                return 1;
+            }
+            if (o1.getTags().size() != 0 && o2.getTags().size() == 0) {
+                return -1;
+            }
+            if (o1.getTags().size() == 0 && o2.getTags().size() == 0) {
+                return o1.getName().fullName.compareToIgnoreCase(o2.getName().fullName);
+            }
+            return o1.getTags().iterator().next().tagName
+                    .compareToIgnoreCase(o2.getTags().iterator().next().tagName);
+        }
+    };
```

Comment:
Additionally, consider creating a `PersonsComparator` class which stores all these comparators as public static constants so that both sort commands can refer to the same `Comparator` object.